### PR TITLE
Feature/upgrade node and fix ports

### PR DESCRIPTION
**Pull request by apricot13**
* fixes issue on local dev env https://github.com/wearefuturegov/outpost-api-service/issues/21 local dev port is now `7000`
* updates node version to `16.18.0` from `14.17.1`
* started documenting the main controller for writing tests as part of testing all the endpoints weren't erroring
* I have run the create indices locally on a new db and it didn't error, but no tests to confirm this on staging
* had to change the ssl package
* setup dev environments as the above package was different, the old one didn't touch development env this one does so I updated the `npm run dev` command to include `NODE_ENV`
* fixed two changes to mongoose that had changed since the versions changed

Test url is here: https://hannahs-outpost-tests-api.herokuapp.com/api/v1/services (currently pointing to greenwich mongodb)

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ npm i
 npm run dev
 ```
 
-By default it will be on `localhost:5000/api/v1/services`.
+By default it will be on `localhost:7777/api/v1/services`.
```
Comment (em-cd) [inline readme.md]: Should this say 7000?

File: controllers/index.js [hunk 2]
```diff
@@ -20,11 +31,14 @@ module.exports = {
         } else {
           query.$text = { $search: req.query.keywords }
         }
+        return query
```
Comment (em-cd) [inline controllers/index.js]: This is making the response hang for me!

**Review by em-cd — CHANGES_REQUESTED** [review]
I'm getting this error:

`Skip value must be non-negative, but received: -9223372036854775808`

when I visit http://localhost:7000/api/v1/services - do you get the same? I think it's to do with the `skip` method on line 152. Not sure why it's only erroring now though, we haven't made any changes to that line in ages!

**Comment by apricot13** [comment]
@em-cd try `nvm use` to make sure your on the right node version and `npm install` as well that should get rid of those issues

**Comment by em-cd** [comment]
> @em-cd try `nvm use` to make sure your on the right node version and `npm install` as well that should get rid of those issues

Just checked, I'm on node 16.18.0 and have run `npm install` and I'm still getting those errors.

File: readme.md (hunk 1 above)
Comment (apricot13) [reply readme.md]: all fixed :)

File: controllers/index.js (hunk 2 above)
Comment (apricot13) [reply controllers/index.js]: my fault! 🤦

**Review by em-cd — APPROVED** [review]
Looks good and all working for me now 🎉

**Pull request merged**